crates/rmf_site_picking/src/select/events.rs

```diff
@@ -23,15 +23,25 @@ pub struct Hover(pub Option<Entity>);
 
 /// Used as an event to command a change in the selected entity.
 #[derive(Default, Debug, Clone, Copy, Deref, DerefMut, Event)]
-pub struct Select(pub Option<SelectionCandidate>);
+pub struct Select {
+    #[deref]
+    pub selection_candidate: Option<SelectionCandidate>,
+    pub multi_select: bool,
+}
 
 impl Select {
-    pub fn new(candidate: Option<Entity>) -> Select {
-        Select(candidate.map(|c| SelectionCandidate::new(c)))
+    pub fn new(candidate: Option<Entity>, multi_select: bool) -> Select {
```

Comment:
I think we can cut down on a lot of unnecessary changes in this PR by tweaking this API:

```rust
pub fn new(candidate: Option<Entity>) -> Self {
    Self {
        candidate,
        multi_select: false,
    }
}

pub fn multi_select(mut self, on: bool) -> Self {
    self.multi_select = on;
    self
}
```

This will allow us to revert the changes that were made to many of the existing uses of `Select::new`.

Reply:
Great idea!